static/app/components/charts/chartZoom.tsx

```diff
@@ -338,4 +338,257 @@ class ChartZoom extends Component<Props> {
   }
 }
 
+/**
+ * This hook provides an alternative to using the `ChartZoom` component. It returns
+ * the props that would be passed to the `BaseChart` as zoomRenderProps.
+ */
+export function useChartZoom({
```

Comment:
Can we move this to its own file?

Should we deprecate `<ChartZoom>` and convert our other usages of it if it has perf problems?